### PR TITLE
add jacrev docstring

### PR DESCRIPTION
**Pull request by samdow**
Adds docstring for jacrev for #224, also does some minor cleanup of the readme in the `jacrev` section and removes a broken example

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -192,15 +194,6 @@ batched jacobians:
...
 ```
 
-`jacrev` can be composed with itself to produce hessians:
-```py
->>> def f(x):
->>>   return x.sin().sum()
->>>
->>> x = torch.randn(5)
->>> hessian = jacrev(jacrev(f))(x)
-```
```
Comment (samdow) [inline README.md:202]: I like this example but it's currently broken, erroring that reshape isn't supported (through a fallback or as a prim). We can get it to work with just sum, but then the hessian is all 0s and that's not super interesting

File: functorch/_src/eager_transforms.py [hunk 2]
```diff
@@ -173,7 +174,73 @@ def wrapper(cotangents, retain_graph=True, create_graph=None):
...
+        >>>   return x + y ** 2
+        >>>
+        >>> x, y = torch.randn(5), torch.randn(5)
+        >>> jacobian = jacrev(f, argnums=(0,1))(x, y)
+        >>> expectedX = torch.diag(torch.ones_like(x))
+        >>> expectedY = torch.diag(2 * y)
+        >>> assert torch.allclose(jacobian[0], expectedX)
+        >>> assert torch.allclose(jacobian[1], expectedY)
+
+    .. note:
```
Comment (samdow) [inline functorch/_src/eager_transforms.py:229]: Since the function calls vmap and grad, figured this note also applied to `jacrev`

File: README.md [hunk 3]
```diff
@@ -169,13 +169,15 @@ When composed with `vmap`, `grad` can be used to compute per-sample-gradients:
...
+The `vjp` transform applies `func` to `inputs` and returns a new function that
+computes vjps given some `cotangents` Tensors.
 ```
 >>> from functorch import vjp
 >>> outputs, vjp_fn = vjp(func, inputs); vjps = vjp_fn(*cotangents)
 ```
-The `vjp` transform applies `func` to `inputs` and returns a new function that
-computes vjps given some `cotangents` Tensors.
 
+The `jacrev` transform returns a new function that takes in `x` and returns the
```
Comment (samdow) [inline README.md:179]: this just fixes it so all descriptions of what's about to happen come before the code example

File: README.md (hunk 1 above)
Comment (Chillee) [reply README.md:202]: wait, reshape isn't supported? That seems like something we should fix :P I'll take a look.

Comment (samdow) [reply README.md:202]: Oh wait this might have been an old version--I pulled main and it looks like it's working okay. Sorry about the false alarm
